libs/ui-lib/lib/ocm/components/clusterConfiguration/networkConfiguration/AdvancedNetworkFields.tsx

```diff
@@ -37,57 +37,80 @@ const AdvancedNetworkFields = () => {
 
   const isDualStack = values.stackType === DUAL_STACK;
 
+  // Keep latest arrays in refs so we can avoid re-running effect on every array edit
+  const clusterNetworksRef = React.useRef(values.clusterNetworks);
+  const serviceNetworksRef = React.useRef(values.serviceNetworks);
+  React.useEffect(() => {
+    clusterNetworksRef.current = values.clusterNetworks;
+  }, [values.clusterNetworks]);
+  React.useEffect(() => {
+    serviceNetworksRef.current = values.serviceNetworks;
+  }, [values.serviceNetworks]);
+
+  // Extract to stable var for dependency list
+  const primaryMachineNetworkCidr = React.useMemo(
+    () => values.machineNetworks?.[0]?.cidr,
+    [values.machineNetworks],
+  );
```

Comment:
the memo here has no purpose
```suggestion
  const primaryMachineNetworkCidr = values.machineNetworks?.[0]?.cidr
  );
```